application/src/main/java/com/sanctionco/thunder/dao/dynamodb/DynamoDbUsersDao.java

```diff
@@ -112,45 +137,67 @@ public User update(@Nullable String existingEmail, User user) {
       return updateEmail(existingEmail, user);
     }
 
-    // Compute the new data
-    long now = Instant.now().toEpochMilli();
-    String newVersion = UUID.randomUUID().toString();
-    String document = UsersDao.toJson(mapper, user);
-
     // Get the old version
-    Item item;
+    GetItemRequest request = GetItemRequest.builder()
+        .tableName(tableName)
+        .key(Collections.singletonMap("email",
+            AttributeValue.builder().s(user.getEmail().getAddress()).build()))
+        .build();
+
+    GetItemResponse response;
     try {
-      item = table.getItem("email", user.getEmail().getAddress());
-    } catch (AmazonClientException e) {
+      response = dynamoDbClient.getItem(request);
+    } catch (SdkException e) {
       LOG.error("The database is currently unresponsive.", e);
       throw new DatabaseException("The database is currently unavailable.",
           DatabaseError.DATABASE_DOWN);
     }
 
-    if (item == null) {
+    if (response.item().size() <= 0) {
       LOG.warn("The user {} was not found in the database.", user.getEmail().getAddress());
       throw new DatabaseException("The user was not found.", DatabaseError.USER_NOT_FOUND);
     }
 
-    String oldVersion = item.getString("version");
+    // Compute the new data
+    long now = Instant.now().toEpochMilli();
```

Comment:
Should be use the current date rather than the Epoch time? Would that be more useful to us?

Reply:
@NickEckert yeah, it actually is the current date just converted to milliseconds. `Instant.now()` is the current datetime, and then `.toEpochMilli()` converts the current datetime into milliseconds since the epoch time.

Thanks for checking though!